.github/workflows/main.yml

```diff
@@ -37,30 +37,33 @@ jobs:
   Build-artifacts:
     needs: [tests, pre-commit]
     runs-on: ubuntu-latest
+    outputs:
+      output: ${{ steps.step_zip.outputs.test }}
     steps:
       - uses: actions/checkout@v2
       - name: Set up Python
         uses: actions/setup-python@v2
         with:
           python-version: 3.8
       - name: create zip package
+        id: step_zip
+        env:
+          file_hash: ${{ hashFiles('api/**', 'requirements.txt') }}
         run: |
           pip3 install -r requirements.txt --target dependencies
-          cd ./dependencies && zip -r9 ../api.zip .
+          cd ./dependencies && zip -r9 ../$file_hash .
           # add API files to Zip file
-          cd ../api && zip -g ../api.zip -r .
-      - name: Upload zip file artifact
-        uses: actions/upload-artifact@v2
-        with:
-          name: api
-          path: api.zip
+          cd ../api && zip -g ../$file_hash -r .
+
+          echo "output=$file_hash" >> $GITHUB_OUTPUT
```

Comment:
for step/job outputs use names that actually describe whats being stored
**file_hash** is fine